daemon/container_unix.go

```diff
@@ -728,7 +728,9 @@ func (container *Container) updateNetworkSettings(n libnetwork.Network) error {
 	if container.NetworkSettings == nil {
 		container.NetworkSettings = &network.Settings{Networks: make(map[string]*network.EndpointSettings)}
 	}
-
+	if !container.hostConfig.NetworkMode.IsHost() && runconfig.NetworkMode(n.Type()).IsHost() {
```

Comment:
We should not be needing this extra check. The check for `IsPrivate()` should disallow connecting to host. Are you able to connect to host after the container is started on a private network?